lib/twitter_cldr/js/mustache/tokenizers/pattern_tokenizer.coffee

```diff
@@ -0,0 +1,24 @@
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+class TwitterCldr.PatternTokenizer
+  constructor : (@data_reader, @tokenizer) ->
+
+  tokenize : (pattern) ->
+    @tokenizer.tokenize(@expand(pattern))
+
+  expand : (pattern) ->
+    # if typeof(pattern) is "string"
+    #   path = (pattern + "").split(".")
+    #   data = (if @data_reader? then @data_reader.pattern_at_path(path) else null)
+    #   next_pattern = (if data["pattern"]? then data["pattern"] else data)
+    #   expand_pattern(next_pattern)
+    # else
+    # TODO - figure thsi out.
```

Comment:
Hmm that's a good question... I actually just had this problem a few days ago when I tried to port some of this data to json (which also doesn't support symbols). I ended up adding a colon to the beginning of each symbol value, eg. `:foo.bar.baz`, then looking at the first character to determine if it represented a redirect or not.

You also might consider pre-processing all the format patterns in Ruby-land instead of doing it in Javascript, since you'll be able to tell what's a symbol and what isn't more easily. That's what we're doing currently for date/time formats and a few other components. You can then inject all the tokens into the mustache instead of simply the un-tokenized patterns.